From ee264da1aa222b354cee36184ab48c2072f62243 Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@cryptomilk.org>
Date: Wed, 23 Jan 2019 11:20:31 +0100
Subject: [PATCH] cmake: Add WITH_EXAMPLES options

This is turned on by default, but you can disable it, if you just
package cmocka for an opterating system.

Fixes #14

Signed-off-by: Andreas Schneider <asn@cryptomilk.org>
---
 CMakeLists.txt      | 4 +++-
 DefineOptions.cmake | 1 +
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7778039..702be97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,7 +60,9 @@ if (UNIT_TESTING)
     add_subdirectory(tests)
 endif (UNIT_TESTING)
 
-add_subdirectory(example)
+if (WITH_EXAMPLES)
+    add_subdirectory(example)
+endif ()
 
 # pkg-config file
 configure_file(cmocka.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/cmocka.pc)
diff --git a/DefineOptions.cmake b/DefineOptions.cmake
index 5bc1ef3..911ff1d 100644
--- a/DefineOptions.cmake
+++ b/DefineOptions.cmake
@@ -1,5 +1,6 @@
 option(WITH_STATIC_LIB "Build with a static library" OFF)
 option(WITH_CMOCKERY_SUPPORT "Install a cmockery header" OFF)
+option(WITH_EXAMPLES "Build examples" ON)
 option(UNIT_TESTING "Build with unit testing" OFF)
 option(PICKY_DEVELOPER "Build with picky developer flags" OFF)
 
-- 
2.18.1

